BitList.cs

```diff
@@ -0,0 +1,1553 @@
+    ICloneable,
+    IEquatable<BitList>
+{
+    private BitArray bits;
+    bool ICollection<bool>.IsReadOnly { get => false; }
+
+    /// <summary>
+    /// This property return true if the instance has no values.
+    /// </summary>
+    public bool IsEmpty { get => bits.Equals(null) || bits.Equals(new BitArray(0)); }
+
+    /// <summary>
+    /// The amount of bits of the instance.
+    /// </summary>
+    public int Count { get => IsEmpty ? 0 : bits.Length; }
+
+    /// <summary>
+    /// Indexer.
+    /// </summary>
+    public bool this[int index]
+    {
+        get => Get(index);
+        set => Set(index, value);
+    }
+
+    /// <summary>
+    /// Indexer.
+    /// </summary>
+    /// <param name="index">Start index.</param>
+    /// <param name="count">Count of bits.</param>
+    /// <exception cref="ArgumentOutOfRangeException" />
+    public BitList this[int index, int count]
+    {
+        get => Get(index, count);
+        set
+        {
+            if (count == value.Count)
+                Set(index, value);
+            else
+                throw new ArgumentOutOfRangeException();
+        }
+    }
+
+    /// <summary>
+    /// No params ctor.
+    /// </summary>
+    public BitList()
+    {
+        bits = null;
+    }
+
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public BitList(bool bit)
+    {
+        bits = new BitArray(new[] { bit });
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A bit array.</param>
+    public BitList(bool[] bits)
+    {
+        this.bits = new BitArray(bits);
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A byte.</param>
+    public BitList(byte bits)
+    {
+        this.bits = new BitArray(new[] { bits });
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A byte array.</param>
+    public BitList(byte[] bits)
+    {
+        this.bits = new BitArray(bits);
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit integer.</param>
+    public BitList(short bits)
+    {
+        this.bits = new BitArray(BitConverter.GetBytes(bits));
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 16 bit integer array.</param>
+    public BitList(short[] bits)
+    {
+        var bytesList = new List<byte[]>();
+        foreach (var num in bits)
+            bytesList.Add(BitConverter.GetBytes(num));
+        var byteList = new List<byte>();
+        foreach (var bytes in bytesList)
+        {
+            foreach (var num in bytes)
+                byteList.Add(num);
+        }
+        this.bits = new BitArray(byteList.ToArray());
+    }
+
+    /// <summary>
+    /// Ctor.
+    /// </summary>
+    /// <param name="bits">A 32 bit integer.</param>
+    public BitList(int bits)
+    {
+        this.bits = new BitArray(new int[] { bits });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)